src/main/java/org/datadog/jmxfetch/util/CustomLogger.java

```diff
@@ -1,123 +1,166 @@
 package org.datadog.jmxfetch.util;
 
 import lombok.extern.slf4j.Slf4j;
-import org.apache.logging.log4j.Level;
-import org.apache.logging.log4j.LogManager;
-import org.apache.logging.log4j.core.Appender;
-import org.apache.logging.log4j.core.LoggerContext;
-import org.apache.logging.log4j.core.appender.ConsoleAppender;
-import org.apache.logging.log4j.core.appender.RollingFileAppender;
-import org.apache.logging.log4j.core.appender.rolling.DefaultRolloverStrategy;
-import org.apache.logging.log4j.core.appender.rolling.SizeBasedTriggeringPolicy;
-import org.apache.logging.log4j.core.config.Configuration;
-import org.apache.logging.log4j.core.config.Configurator;
-import org.apache.logging.log4j.core.config.LoggerConfig;
-import org.apache.logging.log4j.core.layout.PatternLayout;
 
+import org.datadog.jmxfetch.util.LogLevel;
+import org.datadog.jmxfetch.util.StdoutConsoleHandler;
+
+import java.text.SimpleDateFormat;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.List;
+import java.util.Locale;
 import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.atomic.AtomicInteger;
+import java.util.logging.ConsoleHandler;
+import java.util.logging.FileHandler;
+import java.util.logging.Handler;
+import java.util.logging.Level;
+import java.util.logging.LogRecord;
+import java.util.logging.Logger;
+import java.util.logging.SimpleFormatter;
 
 
 @Slf4j
 public class CustomLogger {
     private static final ConcurrentHashMap<String, AtomicInteger> messageCount
             = new ConcurrentHashMap<String, AtomicInteger>();
+
     private static final String LAYOUT = "%d{yyyy-MM-dd HH:mm:ss z} | JMX | %-5p | %c{1} | %m%n";
 
     private static final String LAYOUT_RFC3339 =
         "%d{yyyy-MM-dd'T'HH:mm:ss'Z'} | JMX | %-5p | %c{1} | %m%n";
 
-    // log4j2 uses SYSTEM_OUT and SYSTEM_ERR - support both
-    private static final String SYSTEM_OUT_ALT = "STDOUT";
-    private static final String SYSTEM_ERR_ALT = "STDERR";
+    private static final String DATE_JDK14_LAYOUT = "yyyy-MM-dd HH:mm:ss z";
+    private static final String DATE_JDK14_LAYOUT_RFC3339 = "yyyy-MM-dd'T'HH:mm:ssXXX";
+    private static final String JDK14_LAYOUT = "%s | JMX | %2$s | %3$s | %4$s%n";
 
-    /** Sets up the custom logger to the specified level and location. */
-    public static void setup(Level level, String logLocation, boolean logFormatRfc3339) {
-        final LoggerContext ctx = (LoggerContext) LogManager.getContext(false);
-        final Configuration config = ctx.getConfiguration();
+    private static boolean isStdErr(String target) {
+        List<String> stderrs = Arrays.asList("SYSTEM.ERR", "SYSTEM_ERR", "STDERR");
```

Comment:
nit: Should the `Arrays.asList` in these functions be `static` themselves? I'm guessing either way it's not a hot path and it doesn't matter so it's not a blocker.

Reply:
Yes, since it's not in a hot path I won't change this now, resolving.